proposals/lip-0041.md

```diff
@@ -170,6 +167,88 @@ This function verifies the nonce of a given transaction `trs` and returns a bool
 
 This function returns `true` if the transaction object contains a valid nonce, `false` otherwise. This is done using the `isValidNonce` function defined [below](#transaction-verification).
 
+### Genesis Block Processing
+
+The following two steps are executed as part of the genesis block processing.
+
+#### Genesis State Initialization
+
+During the genesis state initialization stage, the following steps are executed. If any step fails, the block is discarded and has no further effect.
+
+Let `genesisBlockAssetBytes` be the bytes included in the block assets for the Auth module and let `genesisBlockAssetObject` be the deserialization of `genesisBlockAssetBytes` according to the `genesisAuthStoreSchema` schema, given below. 
+
+* Check that `genesisBlockAssetObject.authDataSubstore.storePrefix` equals `STORE_PREFIX_AUTH`.
+* Check that every entry `authAccount` in `genesisBlockAssetObject.authDataSubstore.storeEntries` has a different `authAccount.storeKey`.
+* For each entry `authAccount` in `genesisBlockAssetObject.authDataSubstore.storeEntries`, if `authAccount.storeValue.numberOfSignatures` is not 0 (a multisignature account), check that the rules listed in [LIP 0017](https://github.com/LiskHQ/lips/blob/master/proposals/lip-0017.md#multisignature-account-registration) are respected. In particular:
+  * The keys in the `authAccount.storeValue.mandatoryKeys` array are pairwise distinct and ordered lexicographically. The array may be empty.
+  * The keys in the `authAccount.storeValue.optionalKeys` array are pairwise distinct and ordered lexicographically. The array may be empty.
+  * The arrays `authAccount.storeValue.mandatoryKeys` and `authAccount.storeValue.optionalKeys` are disjoint; their union has at least one element and at most 64 elements.
+  * `authAccount.storeValue.numberOfSignatures` is an integer between 1 and the total number of public keys of the account (sum of the length of `authAccount.storeValue.mandatoryKeys` and `authAccount.storeValue.optionalKeys`).
+  * `authAccount.storeValue.numberOfSignatures` is not smaller than the length of `authAccount.storeValue.mandatoryKeys`.
+* For each entry `authAccount` in `genesisBlockAssetObject.authDataSubstore.storeEntries`, create a corresponding entry in the auth data substore with `storeKey = authAccount.storeKey` and `storeValue = authAccount.storeValue`.
+
+
+```java
+genesisAuthStoreSchema = {
+  "type": "object",
+  "required": ["authDataSubstore"],
+  "properties": {
+    "authDataSubstore": {
+      "type": "object",
+      "required": ["storePrefix", "storeEntries"],
+      "properties": {
+        "storePrefix": {
+          "dataType": "bytes"
+        },
+        "storeEntries": {
+          "type": "array",
+          "items": {
+            "storeKey": "bytes",
+            "storeValue": {
+              "type": "object",
```

Comment:
Think we are missing the field Number for `storeValue`.

Reply:
fixed